WIP: merge (mesonified) upstream master into windows-meson

Ericson2314 opened the pull request:
I'll try to chip away at it.

1. [ ] Conflicts
   - [x] ~~src~~
   - [ ] tests
2. [x] Linux build failures
3. [ ] MinGW build failures
   - [x] Can enter shell (`.#nix-windows` / `packages.${system}.nix-windows`)
   - [x] Can run `mesonConfigurePhase`
   - [ ] Can build successfully 

We probably want to get `windows-meson` into more of a known-good state before merging this, too.

abathur committed:
update macOS version handling for Big Sur

Keeping this commit narrow for reviewability, but some of these
conditionals will change in subsequent commits in this PR.

Fixes #3852.

abathur committed:
fix xpath and conditional bugs; xpath -> xmllint

- xpath -> xmllint: xpath's cli interface changed in Big Sur
  rather than add conditional logic for picking the correct
  syntax for xpath, I'm changing to xmllint --xpath, which
  appears to be consistent across versions I've tested...

- /plist/dict/key[text()='Writable']/following-sibling::true[1]
  doesn't do quite what's expected. It was written to try to
  select a <true /> node paired with the Writable key, but it
  will also select the *next* <true /> node that appears even
  if it was paired with another key.

- I think there's also a logic bug in the conditionals here.
  I'm not sure anyone ever actuall saw it, thanks to the xpath
  bug, though. With the xpath fix, this conditional passes if /nix
  does not exist, / IS writable, and the version is Catalina+.

  I think it meant to test for /nix does not exist, / is NOT
  writable, and the version is Catalina+. I reworked this lightly
  to make it a little clearer at the code level.

abathur committed:
replace xpath with xmllint --xpath; simplify

As mentioned in previous commit, Big Sur changes the syntax for the
xpath command slightly.

In the process of testing out replacements for these, I noticed a few
small simplification wins.

abathur committed:
adapt to apfs.util flag diff in catalina/big sur

Fixes #3957. Just runs both forms to minimize moving parts.

abathur committed:
restore create-darwin-volume to release tarball

The move from release.nix to flake.nix appears to have lost some
changes from #3628 / 1c56f18a8122b605c28000e295d5e223f272cccd, leaving
create-darwin-volume.sh out of the release tarball.

Under the assumption that this was just an accident/byproduct of when
flake.nix split off and not intentional, I am restoring those edits.

abathur committed:
fix skipped multi-user install steps on macOS

Some of the changes in #3788 to support non-systemd Nix installs
don't appear to be aware that the darwin installer exists, which
resulted in some skipped steps and inappropriate instructions.

abathur committed:
create missing profile files to fix zsh envvars

Env vars for ZSH were moved from /etc/zshrc to /etc/zshenv in #3608
to address an issue with zshrc getting clobbered by OS updates, but
/etc/zshenv doesn't exist by default--so *nothing* would get set up
for zsh users unless they already happened to have /etc/zshenv.

Creating these files if they don't exist. Also cut separate creation
of profile.d/nix.sh, which isn't needed now.

thufschmitt committed:
Allow non-CA derivations to depend on CA derivations

thufschmitt committed:
Inline `unkownHashes`

See https://github.com/NixOS/nix/pull/4056#discussion_r493661632

0mp committed:
Fix tar invocation on FreeBSD

tar(1) on FreeBSD does not use standard output or input when the -f flag
is not provided. Instead, it defaults to /dev/sa0 on FreeBSD.

Make this tar invocation a bit more robust and explicitly tell tar(1) to
use standard output.

This is one of the issues discovered while porting Nix to FreeBSD. It has
been tested and committed locally to FreeBSD ports:
https://svnweb.freebsd.org/ports/head/sysutils/nix/Makefile?revision=550026&view=markup#l108

Ericson2314 committed:
Update src/libstore/binary-cache-store.cc

Co-authored-by: Eelco Dolstra <edolstra@gmail.com>

Ericson2314 committed:
Merge remote-tracking branch 'upstream/master' into ca-floating-upstream

Ericson2314 committed:
Use `drvPath2` and give it a better name

kquick committed:
Fix spelling error.

kquick committed:
Update description of FlakeRef, incorporating suggestion.

kquick committed:
Clarification in the description of the FlakeInput.

kquick committed:
Switch comment format from '// ...' to '/* ... */' for consistency.

domenkozar committed:
Update .github/workflows/test.yml

domenkozar committed:
Update .github/workflows/test.yml

Ericson2314 committed:
Clarify comment a bit

edolstra committed:
Merge pull request #4085 from 0mp/patch-2

Fix tar invocation on FreeBSD

edolstra committed:
Merge pull request #3958 from obsidiansystems/ca-floating-upstream

CA derivations that depend on other CA derivations

edolstra committed:
Merge branch 'github-api-token' of https://github.com/imalsogreg/nix

domenkozar committed:
Merge pull request #4084 from NixOS/dependabot/github_actions/cachix/install-nix-action-v11

Bump cachix/install-nix-action from v10 to v11

edolstra committed:
Cleanup

edolstra committed:
Make Headers an optional argument

edolstra committed:
Merge branch 'master' of github.com:NixOS/nix

kquick committed:
Merge branch 'master' into access-tokens

edolstra committed:
nix registry list: Show 'dir' attribute

Issue #4050.

edolstra committed:
Merge branch 'kwq/path-dir-query' of https://github.com/kquick/nix

kquick committed:
Remove github-access-token in favor of access-token.

Ericson2314 committed:
Revert "Use template structs instead of phantoms"

This reverts commit 9ab07e99f527d1fa3adfa02839da477a1528d64b.

Ericson2314 committed:
nix::worker_proto -> worker_proto

Ericson2314 committed:
Merge remote-tracking branch 'upstream/master' into templated-daemon-protocol

edolstra committed:
Merge branch 'access-tokens' of https://github.com/kquick/nix

edolstra committed:
Simplify

edolstra committed:
Style

edolstra committed:
Installer: Set a known umask

Fixes #1560, #2377.

DavHau committed:
add more examples to --help of `nix run`

edolstra committed:
Merge pull request #4098 from DavHau/davhau-add-nix-run-examples

add more examples to --help of `nix run`

B4dM4n committed:
Fix profile update in nix command

edolstra committed:
Merge pull request #4101 from B4dM4n/fix-profile-update

Fix profile update in nix command

B4dM4n committed:
Fix macOS sandbox build

edolstra committed:
Merge pull request #3895 from obsidiansystems/templated-daemon-protocol

More templated STL support for the daemon protocol

edolstra committed:
Merge pull request #3935 from obsidiansystems/binary-cache-addToStoreFromDump

Get rid of Hash::dummy from BinaryCacheStore

edolstra committed:
Fix s3:// store

Fixes https://github.com/NixOS/nixos-org-configurations/issues/123.

edolstra committed:
Add missing #pragma once

edolstra committed:
Factor out common showBytes()

edolstra committed:
Remove 'using namespace fetchers'

edolstra committed:
mk/precompiled-headers.mk: Fix clang test

"clang++" includes the string "g++" so this test didn't work
properly. However the separate handling of clang might not be needed
anymore...

edolstra committed:
mk/precompiled-headers.mk: Remove special handling for clang

edolstra committed:
Remove unneeded -lboost_* flags

edolstra committed:
Remove static variable name clashes

This was useful for an experiment with building Nix as a single
compilation unit. It's not very useful otherwise but also doesn't
hurt...

edolstra committed:
Shut up some clang warnings

edolstra committed:
Merge pull request #4102 from B4dM4n/macos-sandbox-build

Fix macOS sandbox build

edolstra committed:
Prevent a deadlock when user namespace setup fails

Observed on Centos 7 when user namespaces are disabled:
DerivationGoal::startBuilder() throws an exception, ~DerivationGoal()
waits for the child process to exit, but the child process hangs
forever in drainFD(userNamespaceSync.readSide.get()) in
DerivationGoal::runChild(). Not sure why the SIGKILL doesn't get
through.

Issue #4092.

thufschmitt committed:
Test the remote caching of non-ca-depending-on-ca derivations

Although the non-resolved derivation will never get a cache-hit (it
doesn't have an output path to query the cache for anyways), we might
get one on the resolved derivation.

Ericson2314 committed:
Remove generic std::optional<T> suppport from worker proto

See comment for rational; I think it's good to leave a comment lest
anyone is tempted to add such a sum-type instance again.

Fixes #4113

edolstra committed:
Merge pull request #4120 from obsidiansystems/no-poly-sum-worker-proto

Remove generic std::optional<T> suppport from worker proto

edolstra committed:
Formatting

edolstra committed:
Serialize exceptions from the sandbox process to the parent

Fixes #4118.

edolstra committed:
Serialize exceptions from the daemon to the client

edolstra committed:
Tweak error messages

edolstra committed:
Doh

edolstra committed:
Support user namespaces being disabled

If max_user_namespaces is set to 0, then don't run the build in a user
namespace.

Fixes #4092.

edolstra committed:
Dynamically disable user namespaces if CLONE_NEWUSER fails

This makes builds work inside nixos-enter.

Fixes #3145.

edolstra committed:
Merge pull request #4121 from NixOS/no-user-namespace

Support user namespaces being disabled

Horki committed:
rust: small patches

edolstra committed:
Merge pull request #4125 from Horki/nix_rust_small_patches

rust: small patches

edolstra committed:
Remove stray 'Title:' from the manual

Closes #4096.

thufschmitt committed:
Add a description in the completion outputs

Make nix output completions in the form `completion\tdescription`.
This can't be used by bash (afaik), but other shells like zsh or fish
can display it along the completion choices

thufschmitt committed:
Add a zsh completion script

Based on @clhodapp's suggestion in https://github.com/spwhitt/nix-zsh-completions/issues/32#issuecomment-705315356 and adapted to use the description of the completions

edolstra committed:
Remove stray DerivationOutputsAndPaths type

edolstra committed:
writeFile(): Add error context to writeFull() failure

Issue #4092.

edolstra committed:
Remove Lazy

This fixes a crash during startup when compiling Nix as a single
compilation unit.

Ericson2314 committed:
Merge remote-tracking branch 'upstream/master' into fix-and-ci-static-builds

edolstra committed:
Merge pull request #4128 from tweag/extended-completions

Add a zsh completion script

edolstra committed:
Completions::add(): Guard against newlines

edolstra committed:
nix develop: Source ~/.bashrc

Fixes #4104.

edolstra committed:
nix develop: Unset $HOSTNAME

This is set to "localhost" by stdenv which is probably not what you
want.

Ericson2314 committed:
Split out `local-fs-store.hh`

This matches the already-existing `local-fs-store.cc`.

Ericson2314 committed:
Split out uds-remote-store.{cc.hh}

Ericson2314 committed:
Rename to hand-hold git (build.hh)

Ericson2314 committed:
Trim build.hh

Ericson2314 committed:
Rename to hand-hold git (build/derivation-goal.cc)

Ericson2314 committed:
Trim build/derivation-goal.cc

Ericson2314 committed:
Rename to hand-hold git (build/goal.cc)

Ericson2314 committed:
Trim build/goal.cc

Ericson2314 committed:
Rename to hand-hold git (build/hook-instance.cc)

Ericson2314 committed:
Trim build/hook-instance.cc

Ericson2314 committed:
Rename to hand-hold git (build/local-store-build.cc)

Ericson2314 committed:
Trim build/local-store-build.cc

Ericson2314 committed:
Rename to hand-hold git (build/substitution-goal.cc)

Ericson2314 committed:
Trim build/substitution-goal.cc

Ericson2314 committed:
Rename to hand-hold git (build/worker.cc)

Ericson2314 committed:
Trim build/worker.cc

Ericson2314 committed:
Rename to hand-hold git (lock.hh)

Ericson2314 committed:
Trim lock.hh

Ericson2314 committed:
Rename to hand-hold git (lock.cc)

Ericson2314 committed:
Trim lock.cc

Ericson2314 committed:
Split out `commonChildInit`

Ericson2314 committed:
Prepare for build/* files

Ericson2314 committed:
Merge branches 'build-1', 'build-2', 'build-3', 'build-4', 'build-5', 'build-6', 'build-7' and 'build-8' into split_build_cc

edolstra committed:
Merge pull request #4135 from obsidiansystems/split-local-fs-store-header

Split out `local-fs-store.hh`

Ericson2314 committed:
Properly type the derivation and substitution goal maps

As a bonus, Worker::removeGoal is less inefficient.

Ericson2314 committed:
Copy {uds-,}remote-store.{cc,hh}

This prepares for the splitting that happens in the next commit.

edolstra committed:
Merge pull request #4136 from obsidiansystems/split-uds-remote-store

Split out uds-remote-store.{cc.hh}

Commit:
Handle amount of disk space saved by hard linking being negative

Fixes bogus messages like "currently hard linking saves 17592186044416.00 MiB".

Ericson2314 committed:
Prepare for build/*.hh headers

Ericson2314 committed:
Change .cc files to use split build headers

Ericson2314 committed:
Trim derivation-goal.hh

Ericson2314 committed:
Rename to hand-hold git (derivation-goal.hh)

Ericson2314 committed:
Rename to hand-hold git (goal.hh)

Ericson2314 committed:
Trim goal.hh

Ericson2314 committed:
Rename to hand-hold git (hook-instance.hh)

Ericson2314 committed:
Trim hook-instance.hh

Ericson2314 committed:
Trim substitution-goal.hh

Ericson2314 committed:
Rename to hand-hold git (substitution-goal.hh)

Ericson2314 committed:
Rename to hand-hold git (worker.hh)

Ericson2314 committed:
Trim worker.hh

Ericson2314 committed:
Merge branches 'split_build_hh-0', 'split_build_hh-1', 'split_build_hh-2', 'split_build_hh-3' and 'split_build_hh-4' into split_build_cc

Ericson2314 committed:
Merge remote-tracking branch 'upstream/master' into split_build_cc

Ericson2314 committed:
Merge remote-tracking branch 'obsidian/split_build_cc' into typed-goal-maps

edolstra committed:
Merge pull request #4141 from volth/patch-11

Handle amount of disk space saved by hard linking being negative

edolstra committed:
Merge branch 'split_build_cc' of https://github.com/obsidiansystems/nix

Ericson2314 committed:
Merge remote-tracking branch 'upstream/master' into typed-goal-maps

Ericson2314 committed:
Add some more docs

edolstra committed:
Create /etc/passwd *after* figuring out the sandbox uid/gid

Fixes build failures like

  # nix log /nix/store/gjaa0psfcmqvw7ivggsncx9w364p3s8s-sshd.conf-validated.drv
  No user exists for uid 30012

Ericson2314 committed:
Merge remote-tracking branch 'upstream/master' into typed-goal-maps

Ericson2314 committed:
`build-static` -> `buildStatic` in Nix's flake

Ericson2314 committed:
Merge remote-tracking branch 'upstream/master' into fix-and-ci-static-builds

Ericson2314 committed:
Add missing include for MAX_PATH

And remove one that we didn't actually need to add

Ericson2314 committed:
Make a better -lz hack

Per the comments, the underlying issue is
https://github.com/libarchive/libarchive/issues/1446, knowing this
allows the hack to be much more targetted.

Ericson2314 committed:
Bump Nixpkgs to hopefully fix linkrot

Ericson2314 committed:
Merge remote-tracking branch 'obsidian/bump-nixpkgs' into fix-and-ci-static-builds

edolstra committed:
Merge pull request #4151 from obsidiansystems/bump-nixpkgs

Bump Nixpkgs to hopefully fix linkrot

aszlig committed:
Convert VM tests to Python

Perl-based tests are deprecated since NixOS 20.03 and subsequently got
removed in NixOS 20.09, which effectively means that tests are going to
fail as soon as we build it with NixOS 20.09 or anything newer.

I've put "# fmt: off" at the start of every testScript, because
formatting with Black really messes up indentation and I don't think it
really adds anything in value or readability for inlined Python scripts.

Signed-off-by: aszlig <aszlig@nix.build>

edolstra committed:
Merge branch 'fix-and-ci-static-builds' of https://github.com/obsidiansystems/nix into master

aszlig committed:
tests: Add names to VM tests

Having vm-test-run-unnamed for all the test derivation doesn't look very
nice, so in order to better distinguish them from their store path,
let's actually give them proper names.

Signed-off-by: aszlig <aszlig@nix.build>

edolstra committed:
Merge pull request #4156 from aszlig/vm-test-python

Convert VM tests to Python

Ericson2314 committed:
Merge remote-tracking branch 'upstream/master' into typed-goal-maps

Ericson2314 committed:
tab -> space

Ericson2314 committed:
Just use `auto` in two places.

Ericson2314 committed:
Explain that `upcast_goal` is still a static cast

roberth committed:
Tests for #3964

SFrijters committed:
doc: nix-shell in pure mode does *not* source user bashrc

roberth committed:
DerivationGoal: only retry if output closure incomplete is only problem

roberth committed:
Typo

Ericson2314 committed:
Merge branch 'fix-and-ci-static-builds' into meson-p01arst0rm

Ericson2314 committed:
Merge remote-tracking branch 'upstream/master' into meson-p01arst0rm

edolstra committed:
Merge pull request #4158 from hercules-ci/issue-3964-substitution-loop

Fix substitution loop #3964, #3534

edolstra committed:
Merge pull request #4157 from SFrijters/doc-nix-shell-pure-bashrc

doc: nix-shell in pure mode does *not* source user bashrc

edolstra committed:
Merge pull request #4143 from obsidiansystems/typed-goal-maps

Properly type the derivation and substitution goal maps

edolstra committed:
Update .gitignore

edolstra committed:
Add some missing clean-files

Ericson2314 committed:
WIP Merge branch 'meson' into windows-meson

Ericson2314 committed:
Fix a few more conflicts

edolstra committed:
nix develop: Add --redirect flag to redirect dependencies

This is primarily useful if you're hacking simultaneously on a package
and one of its dependencies. E.g. if you're hacking on Hydra and Nix,
you would start a dev shell for Nix, and then a dev shell for Hydra as
follows:

  $ nix develop \
    --redirect .#hydraJobs.build.x86_64-linux.nix ~/Dev/nix/outputs/out \
    --redirect .#hydraJobs.build.x86_64-linux.nix.dev ~/Dev/nix/outputs/dev

(This assumes hydraJobs.build.x86_64-linux has a passthru.nix
attribute. You can also use a store path.)

This causes all references in the environment to those store paths to
be rewritten to ~/Dev/nix/outputs/{out,dev}. Note: unfortunately, you
may need to set LD_LIBRARY_PATH=~/Dev/nix/outputs/out/lib because
Nixpkgs' ld-wrapper only adds -rpath entries for -L flags that point
to the Nix store.

edolstra committed:
flake.lock: Update

Flake input changes:

* Updated 'nixpkgs': 'github:NixOS/nixpkgs/3a10a004bb5802d5f23c58886722e4239705e733' -> 'github:NixOS/nixpkgs/ad0d20345219790533ebe06571f82ed6b034db31'

edolstra committed:
Switch to Nixpkgs 20.09

edolstra committed:
Remove buildStatic from checks

checks should be relatively fast, but buildStatic depends on a lot of
stuff that isn't in the binary cache (e.g. musl builds of Git and
Mercurial that we probably don't need since we don't link against
them...).

Ericson2314 committed:
A few more conflicts fixed

edolstra committed:
Merge pull request #4162 from edolstra/nixpkgs-20.09

Switch to Nixpkgs 20.09

edolstra committed:
Merge pull request #4080 from kquick/kwq/flake-int-doc

Add some internal documentation for flake support objects.

edolstra committed:
Fix clang build

abathur committed:
discourage casual Big Sur installs

kampka committed:
Add NIX_CONFIG env var for applying nix.conf overrides

Ericson2314 committed:
Tentatively fix rest of conflicts in libutil

Ericson2314 committed:
More conflict fixes

Ericson2314 committed:
Tentatively fix more easier conflicts

Ericson2314 committed:
Start fixing conflicts in tests

domenkozar committed:
Merge pull request #3996 from abathur/macos_big_sur_fixes

Macos big sur installer fixes

mkenigs committed:
Make bash non-interactive for nix develop --phase

Fix #3975: Currently if Ctrl-C is pressed during a phase, the interactive subshell
is not exited. Removing --rcfile when --phase is present makes bash
non-interactive

edolstra committed:
Merge pull request #4168 from mkenigs/fix-3975

Make bash non-interactive for nix develop --phase

edolstra committed:
Remove conf-file.xml

This was probably revived in a bad merge.

Ericson2314 committed:
Get libnixutil.so building on Linux

Ericson2314 committed:
Fix more libstore build failures

Ericson2314 committed:
Fix more errors

edolstra committed:
Merge pull request #4166 from kampka/nix-conf-env

Add NIX_CONFIG env var for applying nix.conf overrides

knedlsepp committed:
Fix the docs about the new NIX_CONFIG env var

This was accidentally documented as NIX_OPTIONS.

edolstra committed:
nix develop: Handle 'declare -ax' in bash output

Fixes 'nix develop nixpkgs#qpdfview'.

edolstra committed:
Merge pull request #4177 from knedlsepp/fix-NIX_CONFIG-doc

Fix the docs about the new NIX_CONFIG env var

edolstra committed:
Merge pull request #4161 from edolstra/nix-develop-redirects

nix develop: Add --redirect flag to redirect dependencies

Commit:
resolve all failures with theirs

Commit:
restore some lines from the windows branch

Commit:
try to build

AtnNn commented:
I've made some progress building this branch in mingw64 in 257a297af

tnias committed:
Add sha512 to hashAlgo listings in manpages (#4186)

edolstra committed:
Revert "Bump version to 3.0"

This reverts commit 189e6f5e1d949f50ab0b6e5acd25e230d206692d.

After some discussion, it seems better not to bump the major version
number since most of the new features since 2.3 are marked
experimental.

edolstra committed:
Make the prompt used in development shells configurable

edolstra committed:
Move Explicit

edolstra committed:
Fix test

edolstra committed:
Remove edition field

edolstra committed:
BaseSetting<StringMap>::set(): Don't append to previous value

edolstra committed:
isTrivial(): Support trivial lists

edolstra committed:
Merge pull request #4056 from tweag/non-ca-depending-on-ca

Allow non-CA derivations to depend on CA ones

Ericson2314 committed:
Hand-migrate a few diffs from build.cc

Ericson2314 committed:
semi-manually Apply some changes from build.cc

Ericson2314 committed:
Manually fix a conflict

Ericson2314 committed:
Manually fix a few more

Ericson2314 committed:
Manually fix a few more 2

Ericson2314 committed:
Manually apply rest of diff

Ericson2314 committed:
Remove old build.cc now that changes have been migrated post split

Ericson2314 committed:
Fix an a conflict from before

Ericson2314 committed:
libstore builds on Linux again

Ericson2314 committed:
Tentatively fix more conflicts

Ericson2314 committed:
Fix linux build

The mercurial windows support will just have to be redone, I'm afraid.
It's changed too much with the introduction of libfetchers.

Ericson2314 committed:
Merge remote-tracking branch 'nix-windows/wip/merge-atnnn' into wip/windows-meson-merge-master

Ericson2314 committed:
Merge remote-tracking branch 'nix-windows/windows-meson' into wip/windows-meson-merge-master

Ericson2314 commented:
Got Linux building again, and merged in latest changes from @AtnNn and @volth. (I had to remove some of @AtnNn's quick fixes to keep the Linux build running.)

Ericson2314 committed:
Merge remote-tracking branch 'upstream/master' into meson

Ericson2314 committed:
Remove stray files

Ericson2314 committed:
Merge remote-tracking branch 'nix-windows/meson' into wip/windows-meson-merge-master

Ericson2314 committed:
No x86_32 static nix jobs for now

Fixes #4175

edolstra committed:
Merge pull request #4194 from Ericson2314/skip-bad-static

No x86_32 static nix jobs for now

mkenigs committed:
Capitalize JSON for consistency

AtnNn commented:
A few build errors with mingw: https://gist.github.com/AtnNn/04100b528d4cc07e178507b929a9ad55

But if this builds on Linux I would recommend merging and dealing with that later.

Ericson2314 commented:
Thanks for the log! 

I think I want to at least *attempt* fixing the mingw first :). I need to figure out how flakes / `nix develop` will allow me to have multiple shells, as that isn't clear to me right now.

edolstra committed:
Consistency

Commit:
Alter "wanted:" to "specified:" in hash mismatch output

This makes it even clearer which of the two hashes was specified in the
nix files. Some may think that "wanted" and "got" is obvious, but:
"got" could mean "got in nix file" and "wanted" could mean "want to see in nix file".

Ericson2314 committed:
Can enter MinGW dev shell

Ericson2314 committed:
Meson configure successfully for MinGW

Lowdown is a portability mess; we'll just skip for now.

Ericson2314 committed:
A few misc fixes for MinGW

Ericson2314 committed:
Just don't try to glob on Windows

No glob.h

Ericson2314 commented:
@AtnNn OK I think I did enough that this should be much more straight-forward to contribute too.

I still sorta feel like merging a known-regression is....inauspicious. Do you think it would make sense for us all to just push to this branch?

(Relatedly, since I haven't tried any of this on Windows or with @volth's nixpkgs-windows work, I'm not even sure if `windows-meson` as-is is not a regression.)

edolstra committed:
Merge pull request #4198 from mkenigs/capitalize-JSON

Capitalize JSON for consistency

edolstra committed:
Merge pull request #4199 from stefanjaax/replaceWantedBySpecified

Alter "wanted:" to "specified:" in hash mismatch output

edolstra committed:
Generalize extra-* settings

This removes the extra-substituters and extra-sandbox-paths settings
and instead makes every array setting extensible by setting
"extra-<name> = <value>" in the configuration file or passing
"--<name> <value>" on the command line.

edolstra committed:
Fix assertion failure in tab completion for --option

Ericson2314 commented:
https://github.com/NixOS/nixpkgs/pull/102070 OK at some point we'll be able to switch back to 20.09.

roberth committed:
nix-shell.md: evaluated -> run

Use "run" to avoid confusion with Nix evaluation.
"evaluated" was intended to reference bash eval but it's ambiguous.

roberth committed:
nix-shell.md: Extend shellHook example

edolstra committed:
Merge pull request #4202 from hercules-ci/nix-shell-doc

Nix shell doc

edolstra committed:
Fix test

roberth committed:
Fix memory corruption caused by GC-invisible coroutine stacks

Crucially this introduces BoehmGCStackAllocator, but it also
adds a bunch of wiring to avoid making libutil depend on bdw-gc.

Part of the solutions for #4178, #4200

roberth committed:
Fix RemoteStore pool deadlock in filterSource etc

roberth committed:
BoehmGCStackAllocator: increase stack size to 8MB

The default stack size was not based on the normal stack size and
was too small.

roberth committed:
Restore RestrictedStore.addToStoreFromDump implementation

It was accidentally removed in commit ca30abb3fb36440e5a13161c39647189036fc18f

grahamc committed:
Merge pull request #4209 from hercules-ci/fix-restricted-store-addToStoreFromDump

Restore RestrictedStore.addToStoreFromDump implementation

edolstra committed:
Don't send eval-related settings to the daemon

edolstra committed:
Merge pull request #4207 from hercules-ci/fix-RemoteStore-filterSource-deadlock

Fix RemoteStore pool deadlock in filterSource etc

mkenigs committed:
docs: consistent console prompt (#4213)

Everywhere else a $ is used

edolstra committed:
Don't use readDerivation() in addValidPath()

readDerivation() requires a valid path.

Fixes #4210.

edolstra committed:
nix repl: Fix handling of multi-line expressions

grahamc committed:
nix-copy-closure: verify it works with drvs

Creates test coverage for #4210 and 7cf874c17d466d5cffdb0eb6215fcfe8930ed757

domenkozar committed:
Merge pull request #4214 from grahamc/test-nix-copy-closure

nix-copy-closure: verify it works with drvs

edolstra committed:
Add FIXME

edolstra committed:
LocalStore: Get rid of recursive_mutex

edolstra committed:
Fix error message 'assertion failed at'

Ericson2314 committed:
Merge remote-tracking branch 'nix-windows/windows-meson' into wip/windows-meson-merge-master

Ericson2314 committed:
Uncomment bit requiring a newer Meson

Ericson2314 committed:
Merge remote-tracking branch 'upstream/master' into wip/windows-meson-merge-master

edolstra committed:
Merge pull request #4206 from hercules-ci/fix-coroutine-gc

Fix memory corruption caused by GC-invisible coroutine stacks

Ericson2314 committed:
Merge remote-tracking branch 'upstream/master' into meson

Ericson2314 committed:
Add missing libexpr dep

Ericson2314 committed:
Merge branch 'meson' into wip/windows-meson-merge-master

Et7f3 commented:
If we want to help get native nix on windows wich version should we choose ? source or dest branch ? I am interrested in mingw setup since it seem easier to setup.

Ericson2314 commented:
I've been waiting to return to this until https://github.com/NixOS/rfcs/pull/132 is resolved.
